File: Python/008.py
import sys
import os
import timeit
import math
os.chdir(os.path.dirname(os.path.abspath(__file__))) 

###############################################################################
# Find the greatest product of five consecutive digits in the 1000-digit number.
###############################################################################


def read_digits(path):
	f = open(path, "r")
	digits = f.read()
	digits = digits.replace("\n", "")
	return list(map(int, list(digits)))


def find_greatest_product(numbers):
	max_product = 0
	for i in range(4, len(numbers)):
		product = numbers[i - 4] * numbers[i - 3] * numbers[i - 2] * numbers[i - 1] * numbers[i]
		max_product = max(max_product, product)

	return max_product


def main():
    input_file = "../data/008_input.txt"
    greatest_product = find_greatest_product(read_digits(input_file))
    print("Greatest product of five consecutive digits: {}".format(greatest_product))


print("Time: {}".format(timeit.timeit('main()', "from __main__ import main", number=1)))

File: Julia/008.jl
include("EulerLib.jl")

###############################################################################
# Find the greatest product of five consecutive digits in the 1000-digit number.
###############################################################################

function myparser(c)
    parseint(string(c))
end

function main()
    s = split(replace(readall("../data/008_input.txt"), '\n', ""), "")
    digits = map((c -> parseint(string(c))), s)
    greatest = 0
    for (i,d) = enumerate(digits)
        if i < 5
            continue
        end
        greatest = max(greatest, digits[i-4] * digits[i-3] * digits[i-2] * digits[i-1] * digits[i])
    end
    println("Greatest product of five consecutive digits: $(greatest)")
end

@time main()

